pkg/osvscanner/osvscanner.go

```diff
@@ -644,72 +644,78 @@ func createCommitQueryPackage(commit string, source string) scannedPackage {
 	}
 }
 
-func scanDebianDocker(r reporter.Reporter, dockerImageName string) ([]scannedPackage, error) {
-	cmd := exec.Command("docker", "run", "--rm", "--entrypoint", "/usr/bin/dpkg-query", dockerImageName, "-f", "${Package}###${Version}\\n", "-W")
-	stdout, err := cmd.StdoutPipe()
+func runCommandLogError(r reporter.Reporter, name string, args ...string) error {
+	cmd := exec.Command(name, args...)
 
+	// Get stderr for debugging when docker fails
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		r.Errorf("Failed to get stderr: %s\n", err)
+		return err
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		r.Errorf("Failed to run docker command (%q): %s\n", cmd.String(), err)
+		return err
+	}
+	// This has to be captured before cmd.Wait() is called, as cmd.Wait() closes the stderr pipe.
+	var stderrLines []string
+	scanner := bufio.NewScanner(stderr)
+	for scanner.Scan() {
+		stderrLines = append(stderrLines, scanner.Text())
+	}
+
+	err = cmd.Wait()
 	if err != nil {
-		r.Errorf("Failed to get stdout: %s\n", err)
+		r.Errorf("Docker command exited with code (%q): %d\nSTDERR:\n", cmd.String(), cmd.ProcessState.ExitCode())
+		for _, line := range stderrLines {
+			r.Errorf("> %s\n", line)
+		}
+
+		return errors.New("failed to run docker command")
+	}
+
+	return nil
+}
+
+func scanDockerImage(r reporter.Reporter, dockerImageName string) ([]scannedPackage, error) {
+	tempImageFile, err := os.CreateTemp("", "docker-image-*.tar")
+	if err != nil {
+		r.Errorf("Failed to create temporary file: %s\n", err)
 		return nil, err
 	}
-	stderr, err := cmd.StderrPipe()
 
+	err = tempImageFile.Close()
 	if err != nil {
-		r.Errorf("Failed to get stderr: %s\n", err)
 		return nil, err
 	}
+	defer os.Remove(tempImageFile.Name())
 
-	err = cmd.Start()
+	r.Infof("Pulling docker image (%q)...\n", dockerImageName)
+	err = runCommandLogError(r, "docker", "pull", "-q", dockerImageName)
```

Comment:
What's the CLI output if `docker` doesn't exist on the user's system?

Reply:
```
            Failed to run docker command ("docker pull -q alpine:non-existent-tag"): exec: "docker": executable file not found in $PATH
            exec: "docker": executable file not found in $PATH
```

Reply:
nice! that's pretty clear to the user.